### PR TITLE
feat: default snapshot behaviour is now to load if it exists

### LINKED ISSUE #5442: Always load from latest local snapshot as default behaviour
# Task Overview

**In order to** have vega's snapshot system work more intuitively
**We will** change the default behaviour when starting a node
**So that** it will always load from the latest local snapshot if one exists

See https://vegaprotocol.slack.com/archives/CAHA5EX0F/p1654603454810099

Basically the most common usecase for when you've stopped you node and want to start it again, is to load from where you got up to. This is most important when a node was started using state-sync and the tendermint block store does not contain enough history to replay the chain from zero, any attempt to replay from zero in this case will *always* fail so it makes no sense for it to be the default.

Beware that a system-test may need updating.

# Acceptance Criteria
How do we know when this technical task is complete:
- It is possible to start a vega node from the lastest snapshot without having to change any configuration

# Dependencies
Links to any tickets that have a dependant relationship witht his task.

# Additional Details (optional)
Any additional information including known dependencies, impacted components.

# Examples (optional)
Code snippets, API calls that could be used on dependant tasks.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Create relevant for [system-test](https://github.com/vegaprotocol/system-tests/issues) tickets with feature labels
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All acceptance criteria confirmed to be met, or, reasons why not discussed with the engineering leadership team
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new system-tests
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by wwestgarth**
closes #5442 

I've kicked off a full run, which is its usual shade of red, but none of the snapshot tests or snapshot heavy network-infra-validator tests are failing, so we're good: https://jenkins.ops.vega.xyz/blue/organizations/jenkins/common%2Fsystem-tests/detail/system-tests/9222/pipeline/33/

### CODE REVIEW

File: core/processor/abci.go [hunk 1]
```diff
@@ -575,14 +575,6 @@ func (app *App) OnInitChain(req tmtypes.RequestInitChain) tmtypes.ResponseInitCh
 	ctx = vgcontext.WithTraceID(ctx, hash)
 	app.blockCtx = ctx
 
-	// Start the snapshot engine letting it clear any pre-existing state so that if we are
```
Comment (jeremyletang) [inline core/processor/abci.go:578]: Is that not applicable anymore if we decide to replay explicitly even if we have local snapshots?

Comment (wwestgarth) [reply core/processor/abci.go:578]: It is, I've just moved it here:
```
	if startHeight == 0 {
		// starting a new chain or replaying, not loading snapshot
		// forced chain replay, we need to remove all old snapshots and start again
		e.initialised = false
		e.db.Close()
		e.ClearAndInitialise()
		return false, nil
	}
```

Comment (jeremyletang) [reply core/processor/abci.go:578]: Makes senses, I was too impatient to finish reading it.

Comment (wwestgarth) [reply core/processor/abci.go:578]: Although I'm no longer checking for the error! I'll fix that.

Comment (jeremyletang) [reply core/processor/abci.go:578]: fair enough.

**Review by jeremyletang — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**